gdal/frmts/netcdf/netcdflayersg.cpp

```diff
@@ -0,0 +1,215 @@
+    int cont_id = sg->getContainerId();
+    nccfdriver::SGeometry_PropertyScanner pr(ncid, cont_id);
+    OGRwkbGeometryType owgt = nccfdriver::RawToOGR(sg->getGeometryType(), sg->get_axisCount());
+    
+    if(sg->getGridMappingVarID() != nccfdriver::INVALID_VAR_ID)
+        SetProjectionFromVar(ncid, nc_basevarId, true, sg->getGridMappingName().c_str());     
+
+    // Geometry Type invalid, avoid further processing
+    if(owgt == wkbNone)
+    {
+        throw nccfdriver::SG_Exception_BadFeature();
+    }
+
+    char baseName[NC_MAX_CHAR + 1];
+    memset(baseName, 0, NC_MAX_CHAR + 1);
+    nc_inq_varname(ncid, nc_basevarId, baseName);
+
+    OGRSpatialReference * poSRS = nullptr;
+/*  // Disabled, until SRS function is refactored and fixed
+    if(pszCFProjection != nullptr)
+    {
+        poSRS = new OGRSpatialReference();
+        if(poSRS->importFromWkt(pszCFProjection) != OGRERR_NONE)
+        {
+            delete poSRS;
+            throw nccfdriver::SG_Exception_General_Malformed("SRS settings");
+        }
+
+        poSRS -> Release();
+    }
+*/   
+    netCDFLayer * poL = new netCDFLayer(this, ncid, baseName, owgt, poSRS); 
+
+    poL->EnableSGBypass();
+    OGRFeatureDefn * defn = poL->GetLayerDefn();
+    defn->SetGeomType(owgt);
+
+    size_t shape_count = sg->get_geometry_count();
+
+    // Add properties
+    std::vector<int> props = pr.ids();
+    for(size_t itr = 0; itr < props.size(); itr++)
+    {
+        poL->AddField(props[itr]);    
+    }
+
+    for(size_t featCt = 0; featCt < shape_count; featCt++)
+    {
+        OGRGeometry * geometry;
+
+        try
+        {
+            switch(sg->getGeometryType())
+            {        
+                case nccfdriver::POINT:
+                    geometry = new OGRPoint;
+                    break;
+                case nccfdriver::LINE:
+                    geometry = new OGRLineString;
+                    break;
+                case nccfdriver::POLYGON:
+                    geometry = new OGRPolygon;
+                    break;
+                case nccfdriver::MULTIPOINT:
+                    geometry = new OGRMultiPoint;
+                    break;
+                case nccfdriver::MULTILINE:
+                    geometry = new OGRMultiLineString;
+                    break;
+                case nccfdriver::MULTIPOLYGON:
+                    geometry = new OGRMultiPolygon;
+                    break;
+                default:
+                    throw nccfdriver::SG_Exception_BadFeature();
+                    break;
+            }
+        }
+        // This may be unncessary, given the check previously...
+        catch(nccfdriver::SG_Exception&)
+        {
+            delete poL;
+            throw;
+        }
+
+        int r_size = 0;
+        std::unique_ptr<unsigned char, std::default_delete<unsigned char[]>> wkb_rep = std::unique_ptr<unsigned char, std::default_delete<unsigned char[]>>(sg->serializeToWKB(featCt, r_size));
```

Comment:
can be shortened as:
auto wkb_rep = ...
or
std::unique_ptr<...> wkb_rep(sg->serializeToWKB(featCt, r_size));

Reply:
done